src/reader/parser/mod.rs

```diff
@@ -76,7 +75,7 @@ pub struct PullParser {
     nst: NamespaceStack,
 
     data: MarkupData,
-    finish_event: Option<Result>,
+    terminal_result: Option<Result>,
```

Comment:
Well, it really is terminal -- if that event occurs nothing is to be analysed with this instance of `PullParser`, isn't it?

If I've just persuaded you -- please use `f0410d5` to merge;
Otherwise -- consider merging `75d9f40`.